File: examples/collaborative-assistant/compliance-agent/compliance-agent.js
const { Client } = require('worldlink-client');

async function main() {
    const client = new Client();
    await client.connect();

    const agentId = "compliance_1";
    const manifest = {
        "name": "Compliance Agent",
        "version": "1.0.0",
        "description": "Performs a quality assurance check on the output.",
        "capabilities": ["check_quality"]
    };
    await client.registerAgent(agentId, manifest);
    console.log(`Compliance Agent '${agentId}' registered.`);

    client.subscribe("marketing_plan_generated", async (payload) => {
        const marketingPlan = payload.marketingPlan;
        if (marketingPlan) {
            const qualityCheckResult = "Marketing plan looks good."; // Simplified quality check
            const responseTopic = "marketing_plan_reviewed";
            const responseMessage = { qualityCheck: qualityCheckResult, marketingPlan: marketingPlan };
            await client.publishMessage(responseTopic, responseMessage, agentId);
            console.log(`Compliance Agent '${agentId}' reviewed marketing plan and published result to '${responseTopic}'.`);
        }
    });

    await new Promise(resolve => setTimeout(resolve, 30000));
    await client.disconnect();
}

main();

File: examples/collaborative-assistant/compliance-agent/compliance_agent.py
import asyncio
from worldlink_client import Client

async def main():
    client = Client()
    await client.connect()

    agent_id = "compliance_1"
    manifest = {
        "name": "Compliance Agent",
        "version": "1.0.0",
        "description": "Performs a quality assurance check on the output.",
        "capabilities": ["check_quality"]
    }
    await client.register_agent(agent_id, manifest)
    print(f"Compliance Agent '{agent_id}' registered.")

    async def on_message(topic, payload):
        if topic == "marketing_plan_generated":
            marketing_plan = payload.get("marketing_plan")
            if marketing_plan:
                quality_check_result = "Marketing plan looks good." # Simplified quality check
                response_topic = "marketing_plan_reviewed"
                response_message = {"quality_check": quality_check_result, "marketing_plan": marketing_plan}
                await client.publish_message(response_topic, response_message, sender_id=agent_id)
                print(f"Compliance Agent '{agent_id}' reviewed marketing plan and published result to '{response_topic}'.")

    client.subscribe("marketing_plan_generated", on_message)

    await asyncio.sleep(30)
    await client.disconnect()

if __name__ == "__main__":
    asyncio.run(main())